test/distributed/org/apache/cassandra/distributed/test/auth/MutualTlsCertificateValidityPeriodTest.java

```diff
@@ -0,0 +1,329 @@
+    }
+
+    @Test
+    public void testCertificateReachingMaxValidityPeriod() throws Exception
+    {
+        LOGGER.info("[test step : @Test] testCertificateReachingMaxValidityPeriod");
+        Path clientKeystorePath = generateClientCertificate(b -> b.notBefore(Instant.now().minus(3, ChronoUnit.DAYS))
+                                                                  .notAfter(Instant.now().plus(4, ChronoUnit.DAYS)));
+
+        com.datastax.driver.core.Cluster driver = JavaDriverUtils.create(CLUSTER, null, b -> b.withSSL(getSSLOptions(clientKeystorePath)));
+
+        testWithDriver(driver, (Session session) -> {
+            ResultSet clientView = session.execute(new SimpleStatement("SELECT * FROM system_views.clients"));
+            Assertions.assertThat(clientView).isNotNull().isNotEmpty();
+
+            Optional<Row> thisClient = StreamSupport.stream(clientView.spliterator(), false)
+                                                    .filter(row -> "cassandra_ssl_test".equals(row.getString("username")))
+                                                    .findFirst();
+
+            Assertions.assertThat(thisClient).isPresent();
+            Row row = thisClient.get();
+            Map<String, String> authenticationMetadata = row.getMap("authentication_metadata", String.class, String.class);
+
+            Assertions.assertThat(authenticationMetadata).isNotNull().hasSize(1)
+                      .containsKey("identity")
+                      .extractingByKey("identity", as(InstanceOfAssertFactories.STRING)).isEqualTo(IDENTITY);
+            Assertions.assertThat(row.getString("authentication_mode")).isEqualTo("MutualTls");
+        });
+    }
+
+    @Test
+    public void testFailsWhenCertificateExceedsMaxAllowedValidityPeriod() throws Exception
+    {
+        LOGGER.info("[test step : @Test] testFailsWhenCertificateExceedsMaxAllowedValidityPeriod");
+        Path clientKeystorePath = generateClientCertificate(b -> b.notAfter(Instant.now().plus(365, ChronoUnit.DAYS)));
+
+        com.datastax.driver.core.Cluster driver = JavaDriverUtils.create(CLUSTER, null, b -> b.withSSL(getSSLOptions(clientKeystorePath)));
+
+        try
+        {
+            testWithDriver(driver, null);
+            fail("Should not be able to connect when the certificate exceeds the maximum allowed validity period");
+        }
+        catch (com.datastax.driver.core.exceptions.NoHostAvailableException exception)
+        {
+            Assertions.assertThat(exception)
+                      .hasMessageContaining("The validity period of the provided certificate (527040 minutes) exceeds the maximum allowed validity period of 43200 minutes");
+        }
+    }
+
+    @Test
+    public void testFailsWhenCertificateIsExpired() throws Exception
+    {
+        LOGGER.info("[test step : @Test] testFailsWhenCertificateIsExpired");
+        Path clientKeystorePath = generateClientCertificate(b -> b.notBefore(Instant.now().minus(30, ChronoUnit.DAYS))
```

Comment:
One thing I learned recently is that expired certificates will handshake if the certificate is self signed and directly in the trust store.  I see you are making certificates signed by another CA so this is failing as expected.  Just wondering if you encountered the same :)

Reply:
:+1: (talked out of band, we don't need a test for this, was just sharing an observation :+1:)